### ISSUE TITLE
Plain old React inside of a `WebView`

### ISSUE THREAD

**Issue by vkurchatkin** [facebook/react-native#497]
It would be cool if we could use plain React Components as children of a WebView seamlessly:

```javascript
render()  {
  return (
    <WebView>
      <div>
       <MyReactComponent/>
      </div>
    </WebView>
  );
}

```

Any plans to implement something like this?

**Comment by vjeux** [comment]
That's a super interesting idea. No plans from us but would be cool to explore this idea if you want to :)

**Comment by brentvatne** [comment]
Agreed, I would love to see this

**Comment by pilwon** [comment]
We need #505 and #506 to implement this feature without creating another native module.

**Comment by trusktr** [comment]
This is an amazing idea.

**Comment by ilyabo** [comment]
Would be a game-changer for me!

**Comment by jbhatab** [comment]
+1

**Comment by nicklockwood** [comment]
Note that #505 has now been implemented, so at the very least it should now be possible to load a complete React application into your webView from a local resource file, provided all the JS and CSS are bundled into the same file.

**Comment by pilwon** [comment]
@nicklockwood It appears #505 has not been implemented yet. Please see comment https://github.com/facebook/react-native/issues/505#issuecomment-96841847

**Comment by wlindner** [comment]
I thought it was a cool idea to put React (web) inside of a React Native WebView. It's not nearly as seamless as the original suggestion (I think that would require changes to JSX preprocessing?...) but maybe it will help somebody out:

tutorial: http://wl3.me/2015/05/01/react-in-your-react-native-webview-so-you-can-react-while-you-react.html
repo: https://github.com/wlindner/ReactNativeReactWebview

**Comment by brentvatne** [comment]
Nice one @wlindner :smile:

I'm going to close out this issue as it seems dead and at least for the moment best suited for a community module. Feel free to create another issue if in writing that module you run into an issue with the core infrastructure!

**Comment by DenisIzmaylov** [comment]
The idea is crazy nice!

The problem is in the bridge between React Native thread and WebView thread. And such related things as security, permissions and so on.

So DX in the original post is perfect. Bet on that and would be happy to see it.

**Comment by clovs** [comment]
Hope This feature ;p

**Comment by ethanx94** [comment]
For anyone following this, I had the same idea as well and didn't know there were efforts to get this going. My implementation basically bundles the React app before starting the React Native app. The RN app then references those bundled html/js files. 
I'm having a major issue with this, which is re-rendering the WebView doesn't seem to work. If anyone has any pointers please check out this issue.
https://github.com/ethanx94/reactreactnative/issues/2
